app/models/tag.rb

```diff
@@ -82,7 +82,10 @@ def self.find_preferred_alias_for(word)
 
   def validate_name
     errors.add(:name, "is too long (maximum is 30 characters)") if name.length > 30
-    errors.add(:name, "contains non-alphanumeric characters") unless name.match?(/\A[[:alnum:]]+\z/)
+    # [:alnum] is not used here because it supports diacritical characters.
+    # If we decide to allow diacritics in the future, we should replace the
+    # following regex with [:alnum].
+    errors.add(:name, "contains non-alphanumeric characters") unless name.match?(/\A[[a-z0-9]]+\z/i)
```

Comment:
We should consider changing the wording to "contains non-ASCII characters". Accented characters are alphanumeric too, which is why the `:alnum:` class matches them.